packages/hooks/src/useInputATMFormatter.ts

```diff
@@ -0,0 +1,106 @@
+import { useCallback, useEffect, useMemo, useRef } from 'react';
+import useInputDecimalFormatter from './useInputDecimalFormatter';
+
+const unFormatLocaleStringifyNumber = (input: string, locale: Intl.LocalesArgument) => {
+    const parts = (12345.6789).toLocaleString(locale).match(/(\D+)/g);
```

Comment:
could you please elaborate a bit more on the purpose of this line?

Reply:
shouldn't this be like below?
```suggestion
    const parts = input.toLocaleString(locale).match(/(\D+)/g);
```

Reply:
@mahdiyeh-deriv No it shouldn't be `input`, We use a hardcoded number to figure out the separator character for that locale (`.` or `,`) and then reverse engineer the input to find the number before formating.